.github/workflows/ci.yml

```diff
@@ -37,50 +67,147 @@ jobs:
+      - name: Check with Clippy
+        uses: actions-rs/cargo@v1
         with:
           command: clippy
           args: -- -D warnings
-#      - uses: actions-rs/clippy-check@v1
-#        with:
-#          token: ${{ secrets.GITHUB_TOKEN }}
-#          args: --all-features -- -D warnings
+      - name: Cleanup for caching
+        uses: actions-rs/cargo@v1
+        with:
+          command: clean
+          args: --package dotenv-linter
+        if: steps.cache-target.outputs.cache-hit != 'true'
+  #   - uses: actions-rs/clippy-check@v1
+  #     with:
+  #       token: ${{ secrets.GITHUB_TOKEN }}
+  #       args: --all-features -- -D warnings
 
   build:
     name: Build
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - uses: actions-rs/toolchain@v1
-      with:
-        toolchain: stable
-    - uses: actions-rs/cargo@v1
-      with:
-        command: build
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          profile: minimal
+          toolchain: stable
+      - name: Chown cargo registry because of caching
+        run: sudo chown -R runner ~/.cargo/registry
+      - name: Generate lockfile
+        uses: actions-rs/cargo@v1
+        with:
+          command: generate-lockfile
+      - name: Cache registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: registry-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache target
+        id: cache-target
+        uses: actions/cache@v1
+        with:
+          path: target
+          key: build-target-${{ steps.toolchain.outputs.rustc_hash }}-${{ hashFiles('**/Cargo.lock') }}
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+      - name: Cleanup for caching
+        uses: actions-rs/cargo@v1
+        with:
+          command: clean
+          args: --package dotenv-linter
+        if: steps.cache-target.outputs.cache-hit != 'true'
 
   tests:
     name: Tests and coverage
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v1
       - uses: actions-rs/toolchain@v1
+        id: toolchain
         with:
+          profile: minimal
           toolchain: nightly
           override: true
-      - uses: actions-rs/cargo@v1
+      - name: Chown cargo registry because of caching
+        run: sudo chown -R runner ~/.cargo/registry
+      - name: Generate lockfile
+        uses: actions-rs/cargo@v1
         with:
-          command: clean
-      - uses: actions-rs/cargo@v1
+          command: generate-lockfile
+      # May be there is a better way.
+      - name: Set grcov version
+        run: echo "::set-env name=GRCOV_VERSION::$(cargo search --limit 1 grcov | head -n1 | cut -d '"' -f2)"
+      - name: Cache registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: test-registry-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Why do use the different key `test-registry` instead of `registry`?

Reply:
**Build** job (and other jobs excluding `Tests and coverage`):
```
Downloading crates ...
Downloaded libc v0.2.69
```
**Tests and coverage** job:
```
Downloading crates ...
Downloaded predicates-tree v1.0.0
Downloaded wait-timeout v0.2.0
Downloaded syn v1.0.18
Downloaded predicates v1.0.4
Downloaded doc-comment v0.3.3
Downloaded difference v2.0.0
Downloaded treeline v0.1.0
Downloaded predicates-core v1.0.0
Downloaded assert_cmd v0.12.2
Downloaded escargot v0.5.0
```

Example: [https://github.com/dotenv-linter/dotenv-linter/runs/620343497](https://github.com/dotenv-linter/dotenv-linter/runs/620343497)

Reply:
I see. Thank you.